README.adoc

```diff
@@ -82,7 +88,7 @@ There are two protected branches in `plutus-apps`:
 ** The version of transitive dependencies (`plutus`, `cardano-ledger`, `ouroboros-network`, etc.) should be pinned to the ones from `cardano-node` (or better, `cardano-wallet`)
 ** Changes will be backported form `main` to `next-node`
 * `next-node` branch: should always target the next node release.
-** This branch will eventually be merged in `main` after the Cardano mainnet HF.
+** This branch will eventually be merged in `main` after the Cardano mainnet HF and deleted. Once the next HF is planned, it will be recreated and it will contain an upgraded `cardano-node` version.
```

Comment:
I think that the process is fine but maybe we can do it better?

The branch is temporary, yes. But turns out people still use it to upgrade their code to iterate quickly. And by removing it we can break someone's code. Maybe instead of `next-node` we should name the branch `moving-to-vasil` and each HF with it's own name. We can save these branches forever or for some time, to declare the period of time to give some space to stop using it.

My intention here is to help users not to break their code as they use the commit hashes from different branches. Yes, it's, in some sense, their problem. But I think that we can bring such policies to hardfork branches as they special in that sense and be more user-friendly. Just thoughts.

Also I wanted to propose keeping the log of hardforks to track the changes but then I realised that that's why we want to introduce the CHANGELOG. :)

Reply:
> The branch is temporary, yes. But turns out people still use it to upgrade their code to iterate quickly. And by removing it we can break someone's code. Maybe instead of next-node we should name the branch moving-to-vasil and each HF with it's own name. We can save these branches forever or for some time, to declare the period of time to give some space to stop using it.

The idea is that once a HF happens on mainnet, `next-node` is merged in `main`. Therefore, there's no point in continuing to work with `next-node`. Plus, keeping `next-node` will mean that we need to regularly merge `main` into `next-node`. Therefore, I don't think removing `next-node` will affect anyone since all the changes were merged in `main`. Additionally, this would be transparent to users once we start consistently using versions as releases, then the user wouldn't need to worry about branch conventions (only the contributor).

I'm not against explicitly naming the branch to something like `moving-to-vasil`. I guess the idea of always using the same branch name (`next-node`) would give us some consistency for all HFs so the users know what to expect (instead of having to remember to update the docs at every HF saying the branch name used for the next HF). What do you think?

> My intention here is to help users not to break their code as they use the commit hashes from different branches. Yes, it's, in some sense, their problem. But I think that we can bring such policies to hardfork branches as they special in that sense and be more user-friendly. Just thoughts.

So, as users move away from using commit hashes once we use release consistently, it will solve most of these problems.

> Also I wanted to propose keeping the log of hardforks to track the changes but then I realised that that's why we want to introduce the CHANGELOG. :)

Yep! I was going to bootstrap CHANGELOGs, but people in our team made some compelling arguments for Github releases instead, so we'll need to take some time to note the pros and cons of each and make a decision.

Reply:
>Therefore, I don't think removing next-node will affect anyone since all the changes were merged in main. 

Yeah, I remembered that we use merge commits and as long as we merge the `next-node` into `main` without loosing the commits from `next-node` the users should be fine.

>I'm not against explicitly naming the branch to something like moving-to-vasil. I guess the idea of always using the same branch name (next-node) would give us some consistency for all HFs so the users know what to expect (instead of having to remember to update the docs at every HF saying the branch name used for the next HF). What do you think?

Let's see how it goes, I think, adapting to the community's feedback. 👍

Reply:
Alright then. I'll leave it as is and community feedback would definitely help in this case.